File: 5 - Odysseus' Unbroken Journey/solution.py
def solve(s):
    i, j, ans = 0, 0, 0
    mp = {}
    
    while j < len(s):
        mp[s[j]] = mp.get(s[j], 0) + 1
        
        while mp[s[j]] > 1:
            mp[s[i]] -= 1
            i += 1
        
        ans = max(ans, j - i + 1)
        j += 1
    
    return ans

if __name__ == "__main__":
    t = int(input())
    
    for _ in range(t):
        s = input().strip()
        print(solve(s))

File: 5 - Odysseus' Unbroken Journey/solution.js
function solve(s) {
    let i = 0, j = 0, ans = 0;
    const mp = new Map();
    
    while (j < s.length) {
        mp.set(s[j], (mp.get(s[j]) || 0) + 1);
        
        while (mp.get(s[j]) > 1) {
            mp.set(s[i], mp.get(s[i]) - 1);
            i++;
        }
        
        ans = Math.max(ans, j - i + 1);
        j++;
    }
    
    return ans;
}

const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

readline.question('', (t) => {
    let testCases = parseInt(t.trim());
    
    const processTestCase = () => {
        if (testCases > 0) {
            testCases--;
            readline.question('', (s) => {
                console.log(solve(s.trim()));
                processTestCase();
            });
        } else {
            readline.close();
        }
    };
    
    processTestCase();
});
